torchaudio/datasets/librilight_limited.py

```diff
@@ -2,42 +2,37 @@
 from pathlib import Path
 from typing import List, Tuple, Union
 
+import torchaudio
 from torch import Tensor
 from torch.hub import download_url_to_file
 from torch.utils.data import Dataset
-from torchaudio.datasets.librispeech import load_librispeech_item
+from torchaudio.datasets.librispeech import _get_librispeech_metadata
 from torchaudio.datasets.utils import extract_archive
 
 
 _ARCHIVE_NAME = "librispeech_finetuning"
 _URL = "https://dl.fbaipublicfiles.com/librilight/data/librispeech_finetuning.tgz"
 _CHECKSUM = "5d1efdc777b548194d7e09ba89126e2188026df9fd57aa57eb14408d2b2342af"
+_SUBSET_MAP = {"10min": ["1h/0"], "1h": ["1h/*"], "10h": ["1h/*", "9h"]}
 
 
-def _get_fileids_paths(path, subset, _ext_audio) -> List[Tuple[str, str]]:
+def _get_fileids_paths(path, folders, _ext_audio) -> List[Tuple[str, str]]:
     """Get the file names and the corresponding file paths without `speaker_id`
     and `chapter_id` directories.
     The format of path is like:
         {root}/{_ARCHIVE_NAME}/1h/[0-5]/[clean, other] or
         {root}/{_ARCHIVE_NAME}/9h/[clean, other]
     """
-    if subset == "10min":
-        files_paths = [
-            (os.path.join(os.path.dirname(p), "..", ".."), str(p.stem))
-            for p in Path(path).glob("1h/0/*/*/*/*" + _ext_audio)
-        ]
-    elif subset in ["1h", "10h"]:
-        files_paths = [
-            (os.path.join(os.path.dirname(p), "..", ".."), str(p.stem))
-            for p in Path(path).glob("1h/*/*/*/*/*" + _ext_audio)
+
+    def _get_rel_folder(p, path):
+        rel_path = os.path.relpath(os.path.dirname(p), path)
+        return os.path.dirname(os.path.dirname(rel_path))  # move up 2 levels
+
+    files_paths = []
+    for folder in folders:
+        files_paths += [
+            (_get_rel_folder(p, path), str(p.stem)) for p in Path(path).glob(f"{folder}/*/*/*/*" + _ext_audio)
```

Comment:
Since `p` is already Path object, using the equivalent methods on `p` would make this code simpler. https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

Reply:
I do not see much value on defining the `_get_rel_folder` function inside.
It does not capture a value from local scope, and it's only used once.

Either define the function on module level or do it in the `for` loop.

Also, it's not immediately and intuitively readable what it accomplishes. 
Perhaps change the function name to something more descriptive?